src/network/generator/network-generator.c

```diff
@@ -1220,12 +1234,12 @@ int context_merge_networks(Context *context) {
 
                 r = strv_extend_strv(&network->dns, all->dns, false);
                 if (r < 0)
-                        return r;
+                        return log_oom();
```

Comment:
hmm, log_oom() is on LOG_ERR level. Bu tbelow you log at LOG_DEBUG.

as per coding style we want that functions either log about all errors on their own above LOG_DEBUG, or they do not log about any above LOG_DEBUG (and thus either don#t log at all or stick to LOG_DEBUG) – in the latter case the caller can log instead.

hence decide if this function should log about all errors above LOG_DEBUG or about none, and then stick to it.

log_oom_debug() exists and is your friend btw